integration_tests/src/main/python/fastparquet_compatibility_test.py

```diff
@@ -0,0 +1,324 @@
+    TimestampGen(nullable=False,
+                 start=datetime(2000, 1, 1, tzinfo=timezone.utc),
+                 end=datetime(2200, 12, 31, tzinfo=timezone.utc)),  # Vanilla case.
+    pytest.param(TimestampGen(nullable=False,
+                              start=datetime(1, 1, 1, tzinfo=timezone.utc),
+                              end=datetime(1899, 12, 31, tzinfo=timezone.utc)),
+                 marks=pytest.mark.xfail(reason="fastparquet reads timestamps preceding 1900 incorrectly.")),
+], ids=idfn)
+def test_reading_file_written_with_gpu(spark_tmp_path, column_gen):
+    """
+    This test writes the data-gen output to file via the RAPIDS plugin, then checks that the data is read identically
+    via fastparquet and Spark.
+    There are xfails here because of fastparquet limitations in handling Decimal, Timestamps, Dates, etc.
+    """
+    data_path = spark_tmp_path + "/FASTPARQUET_TEST_GPU_WRITE_PATH"
+
+    gen = StructGen([('a', column_gen),
+                     ('part', IntegerGen(nullable=False))
+                     ], nullable=False)
+    # Write data out with Spark RAPIDS plugin.
+    with_gpu_session(
+        lambda spark: gen_df(spark, gen, 2048).repartition(1).write.mode('overwrite').parquet(data_path),
+        conf=rebase_write_corrected_conf
+    )
+
+    # TODO: Maybe make _assert_equal() available to compare dataframes, regardless of CPU vs GPU?
+    # For now, this compares the results of reading back the GPU-written data, via fastparquet and GPU.
+    assert_gpu_and_cpu_are_equal_collect(read_parquet(data_path), conf=rebase_write_corrected_conf)
+
+
+@pytest.mark.parametrize('column_gen', [
+    ByteGen(nullable=False),
+    ByteGen(nullable=True),
+    ShortGen(nullable=False),
+    ShortGen(nullable=True),
+    IntegerGen(nullable=False),
+    IntegerGen(nullable=True),
+    LongGen(nullable=False),
+    LongGen(nullable=True),
+    FloatGen(nullable=False),
+    FloatGen(nullable=True),
+    DoubleGen(nullable=False),
+    DoubleGen(nullable=True),
+    DecimalGen(nullable=False),
+    DecimalGen(nullable=True),
+    pytest.param(
+        StringGen(nullable=False),
+        marks=pytest.mark.xfail(reason="String columns written with fastparquet are read differently between "
+                                       "Apache Spark, and the Spark RAPIDS plugin. "
+                                       "See https://github.com/NVIDIA/spark-rapids/issues/9387.")),
+    pytest.param(
+        StringGen(nullable=True),
+        marks=pytest.mark.xfail(reason="String columns written with fastparquet are read differently between "
+                                       "Apache Spark, and the Spark RAPIDS plugin. "
+                                       "See https://github.com/NVIDIA/spark-rapids/issues/9387.")),
+    pytest.param(
+        DateGen(nullable=False,
+                start=date(year=2000, month=1, day=1),
```

Comment:
We can use the following min/max as start/end values:
```py
>>> import pandas
>>> pandas.Timestamp.min
Timestamp('1677-09-21 00:12:43.145224193')
>>> pandas.Timestamp.max
Timestamp('2262-04-11 23:47:16.854775807')
```

Reply:
I've adopted this for the date/timestamp limits.